api/src/main/java/org/apache/cloudstack/query/QueryService.java

```diff
@@ -125,6 +125,10 @@ public interface QueryService {
     static final ConfigKey<Boolean> SharePublicTemplatesWithOtherDomains = new ConfigKey<>("Advanced", Boolean.class, "share.public.templates.with.other.domains", "true",
             "If false, templates of this domain will not show up in the list templates of other domains.", true, ConfigKey.Scope.Domain);
 
+    ConfigKey<Boolean> returnVmStatsOnVmList = new ConfigKey<>("Advanced", Boolean.class, "return.vm.stats.on.vm.list", "true",
```

Comment:
Minor nit - the variable could be renamed:
 ```suggestion
    ConfigKey<Boolean> ReturnVmStatsOnVmList = new ConfigKey<>("Advanced", Boolean.class, "return.vm.stats.on.vm.list", "true",
```

See if there's a pattern in gobal setting name we should follow?

Reply:
I'm wondering if the default can be set to `false`, if we do this we'll need to add a note in release notes about page - this breaks the default behaviour but given the listVirtualMachines API is widely used by CAPC, terraform, Ansible and what have you - other integrations could be faster. See https://github.com/apache/cloudstack/issues/8975

Reply:
I would really like to change the default behavior of this and some other APIs. But we currently do not have a proper mechanism to do this. We are slowly starting to discuss such mechanism in https://github.com/apache/cloudstack/discussions/8970#discussion-6554582, and more people discussing and giving ideas would be great.

In any case, since we are already creating a configuration for this, we might as well leave the default behavior as is and let users set this configuration if they feel the need to.

Reply:
Generally speaking, the current mechanism of deprecation and change in behaviour that we've developed and used in recent years is through release notes documentation.